apps/web/src/pages/myprofile/index.tsx

```diff
@@ -0,0 +1,72 @@
+import useUploadImage from "@arcave/components/Modal/common/useUploadImage";
+import ACSkeleton from "@arcave/components/common/Skeleton";
+import MyPage from "@arcave/components/views/MyPage";
+import useAPICategory from "@arcave/services/external/useAPICategory";
+import useAPIUser from "@arcave/services/external/useAPIUser";
+import { useRouter } from "next/router";
+
+export default function MyProfile() {
+  const {
+    loginUser: user,
+    updateUser,
+    withdrawUser,
+  } = useAPIUser({ required: true });
+  const router = useRouter();
+  const { categories } = useAPICategory();
+  const {
+    fileInputRef,
+    handleChangeFileInput,
+    isImageReady,
+    fileImageBlob,
+    previewImageUrl,
+    resetUploadField,
+    handleChangePreviewImageUrl,
+  } = useUploadImage();
+
+  if (!user) {
+    return <ACSkeleton count={1} />;
+  }
+
+  const onClickUserProfileImage = () => {
+    fileInputRef.current?.click();
+  };
+
+  const submit = async (data: any) => {
+    const additional: any = {};
+    if (isImageReady) {
+      additional.userImage = fileImageBlob;
+    }
+    await updateUser({
+      email: user.email!!,
+      nickname: data.nickname,
+      categories: data.categories,
+      ...additional,
+    });
```

Comment:
[제안] 삼항 연산자로 인한 가독성 저하는 고민해볼 문제이긴 하지만, 이렇게 다루면 비교적 절차적인 처리와 객체의 불변성을 지킬 수 있다고 생각해요.

```suggestion
    await updateUser({
      email: user.email!!,
      nickname: data.nickname,
      categories: data.categories,
      ...(isImageReady ? {userImage: fileImageBlob} , {})
    });
```